src/hooks/useImmediateEffect.ts

```diff
@@ -1,4 +1,6 @@
-import { useRef, useEffect, useCallback } from 'react';
+/* eslint-disable react-hooks/exhaustive-deps */
```

Comment:
What would you say to moving this above the dependency array and adding a comment on why we don't want `effect` included. I personally am interested to know why (looks like you can't do `[...changes, effect]` or `changes.concat([effect]);`). It seems like `effect` could potentially get stale if not included in the dependency array.

Reply:
The effect itself can get stale in fact, but this follows React's logic. So we'll only update when the specified effects update not the passed closure. I think since this is quite small I just didn't add it inline 😅